### PR TITLE
ScalaPB support

### PR DESCRIPTION
**Pull request by augi**
A new artifact will be deployed, `grpc-json-bridge-core-scalapb`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,17 @@ val anotherBridge = ReflectionGrpcJsonBridge.createFromServices[Task](s1, s2)
 val jsonResponse = bridge.invoke("com.avast.grpc.jsonbridge.test.TestService/Add", """ { "a": 1, "b": 2} """, Map("My-Header" -> "value"))
 ```
 
+This usage suppose that the project uses standard gRPC Java class generation, using [protoc-gen-grpc-java](https://github.com/grpc/grpc-java/tree/master/compiler).
```
Comment (jakubjanecek) [inline README.md]: suppose -> supposes

File: core-scalapb/src/main/scala/com/avast/grpc/jsonbridge/scalapb/ScalaPBServiceHandlers.scala [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      case Success(m) => Right(m)
+    }
+
+  private def createFutureStubCtor(sd: ServiceDescriptor, inProcessChannel: Channel): () => AbstractStub[_] = {
+    val serviceCompanionClassName = getServiceCompanionClassName(sd)
+    logger.debug(s"Obtaining instance of $serviceCompanionClassName")
+    val serviceCompanionClass = Class.forName(serviceCompanionClassName)
+    val serviceCompanion = serviceCompanionClass.getDeclaredField("MODULE$").get(null)
+    val method = serviceCompanionClass.getDeclaredMethod("stub", classOf[Channel])
+    () =>
```
Comment (jakubjanecek) [inline core-scalapb/src/main/scala/com/avast/grpc/jsonbridge/scalapb/ScalaPBServiceHandlers.scala]: Is there a good reason why this needs to be called later or could you just call it immediately and return the result instead?

Comment (augi) [reply core-scalapb/src/main/scala/com/avast/grpc/jsonbridge/scalapb/ScalaPBServiceHandlers.scala]: The reason is that `Class.forName`, `getDeclaredField` and `getDeclaredMethod` would have significant performance impact.

Comment (jakubjanecek) [reply core-scalapb/src/main/scala/com/avast/grpc/jsonbridge/scalapb/ScalaPBServiceHandlers.scala]: I don't get it. These methods are called anyway so they do have the performance impact. The only thing that is postponed is calling `method.invoke` and I don't see the reason for that.

Comment (augi) [reply core-scalapb/src/main/scala/com/avast/grpc/jsonbridge/scalapb/ScalaPBServiceHandlers.scala]: The point is that `Class.forName`, `getDeclaredField` and `getDeclaredMethod` are very slow and so it's better if they are called just once (as the class is constructed).

The `invoke` method is called every time (of course), and its performance is OK (maybe could be slightly better with `MethodHandle`).

Check e.g. [this fresh article (in Czech)](https://deadbeef.k47.cz/jvm-reflexe.html).

**Pull request merged**